src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextLoader.cs

```diff
@@ -139,6 +139,7 @@ private static bool IsEntryAssembly(Assembly assembly)
             string depsJsonFile = Path.ChangeExtension(assemblyLocation, DepsJsonExtension);
             bool depsJsonFileExists = _fileSystem.File.Exists(depsJsonFile);
 
+#if !NETSTANDARD2_0
```

Comment:
This condition will be true for [netcoreapp build](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyModel/src/Microsoft.Extensions.DependencyModel.csproj#L3) of the library that will be true for NativeAOT, so this is not changing anything for NativeAOT.

Reply:
aah. I miss that. Regarding other comment.

> Single-file and NativeAOT will take the early out based on if (string.IsNullOrEmpty(assemblyLocation)) condition.

I explicitly exception when accessing CodeBase from Silk.Net. Based on the code, I would think the same as you, but I see this exception.
```
CodeBase is not supported on assemblies loaded from a single-file bundle.
 	FSharpWolfenstein.Desktop.exe!S_P_Reflection_Core_System_Reflection_Runtime_Assemblies_RuntimeAssemblyInfo__get_CodeBase() Line 76	Unknown
 	FSharpWolfenstein.Desktop.exe!Microsoft_Extensions_DependencyModel_Microsoft_Extensions_DependencyModel_DependencyContextLoader__GetNormalizedCodeBasePath()	Unknown
 	FSharpWolfenstein.Desktop.exe!Microsoft_Extensions_DependencyModel_Microsoft_Extensions_DependencyModel_DependencyContextLoader__GetDepsJsonPath()	Unknown
 	FSharpWolfenstein.Desktop.exe!Microsoft_Extensions_DependencyModel_Microsoft_Extensions_DependencyModel_DependencyContextLoader__LoadAssemblyContext()	Unknown
 	FSharpWolfenstein.Desktop.exe!Microsoft_Extensions_DependencyModel_Microsoft_Extensions_DependencyModel_DependencyContextLoader__Load()	Unknown
 	FSharpWolfenstein.Desktop.exe!S_P_CoreLib_System_Lazy_1<System___Canon>__ViaFactory() Line 323	Unknown
 	FSharpWolfenstein.Desktop.exe!S_P_CoreLib_System_Lazy_1<System___Canon>__ExecutionAndPublication() Line 347	Unknown
 	FSharpWolfenstein.Desktop.exe!S_P_CoreLib_System_Lazy_1<System___Canon>__CreateValue() Line 435	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_Core_Silk_NET_Core_Loader_DefaultPathResolver__TryLocateNativeAssetInRuntimesFolder()	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_Core_Silk_NET_Core_Loader_DefaultPathResolver___c____cctor_b__23_3()	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_Core_Silk_NET_Core_Loader_DefaultPathResolver__EnumeratePossibleLibraryLoadTargets()	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_Core_Silk_NET_Core_Loader_LibraryLoader__LoadWithResolver()	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_Core_Silk_NET_Core_Loader_LibraryLoader__TryLoadNativeLibrary_2()	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_Core_Silk_NET_Core_Loader_LibraryLoader__TryLoadNativeLibrary_1()	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_Core_Silk_NET_Core_Loader_LibraryLoader__LoadNativeLibrary_1()	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_SDL_Silk_NET_SDL_Sdl__CreateDefaultContext()	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_SDL_Silk_NET_SDL_Sdl__GetApi()	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_Windowing_Sdl_Silk_NET_Windowing_Sdl_SdlPlatform___c____ctor_b__21_0()	Unknown
 	FSharpWolfenstein.Desktop.exe!S_P_CoreLib_System_Lazy_1<Bool>__ViaFactory() Line 323	Unknown
 	FSharpWolfenstein.Desktop.exe!S_P_CoreLib_System_Lazy_1<Bool>__ExecutionAndPublication() Line 347	Unknown
 	FSharpWolfenstein.Desktop.exe!S_P_CoreLib_System_Lazy_1<Bool>__CreateValue() Line 435	Unknown
 	FSharpWolfenstein.Desktop.exe!System_Linq_System_Linq_Enumerable__Any_0<System___Canon>() Line 61	Unknown
 	FSharpWolfenstein.Desktop.exe!Silk_NET_Windowing_Common_Silk_NET_Windowing_Window__Create()	Unknown
 	FSharpWolfenstein.Desktop.exe!FSharpWolfenstein_Desktop_Program__main() Line 197	Unknown
 	FSharpWolfenstein.Desktop.exe!FSharpWolfenstein_Desktop__Module___StartupCodeMain()	Unknown
 	FSharpWolfenstein.Desktop.exe!wmain(int argc, wchar_t * * argv) Line 205	C++
 	[Inline Frame] FSharpWolfenstein.Desktop.exe!invoke_main() Line 90	C++
 	FSharpWolfenstein.Desktop.exe!__scrt_common_main_seh() Line 288	C++

```

Reply:
> Based on the code, I would think the same as you, but I see this exception.

Looks like you are missing https://github.com/dotnet/runtime/pull/48790/. Are you using the latest version of the package?

Reply:
Ooh. My bad. I should try to update version first. Silk.Net use 5.0.0 versions. That explains things. Give me a bit of time to verify, and I will close if that’s true. Sorry to bother

Reply:
As usually you are right. Thanks!